ChronoAPI/ChronoLog/src/log.cpp

```diff
@@ -0,0 +1,108 @@
+//
+// Created by eneko on 12/1/23.
+//
+
+#include "log.h"
+#include <spdlog/spdlog.h>
+
+std::shared_ptr <spdlog::logger> Logger::logger = nullptr;
+
+void Logger::initialize(const std::string &logType, const std::string &location, spdlog::level::level_enum logLevel
+                        , const std::string &loggerName)
+{
+    if(!logger)
+    {
+        if(logType == "file")
+        {
+            logger = spdlog::basic_logger_mt(loggerName, location);
+        }
+        else if(logType == "console")
+        {
+            logger = spdlog::stdout_color_mt(loggerName);
+        }
+        else
+        {
+            throw std::invalid_argument("Invalid log type");
+        }
+
+        logger->set_level(logLevel);
+    }
+    else
+    {
+        throw std::logic_error("Logger is already initialized");
+    }
```

Comment:
1. Return codes are added now. 
2. True. Already modified. 
3. changeConfiguration() is deleted. No needed in the new implementation. For getLogger(), if will imply passing the configurationManager on each call or adding some extra loggic that i do not find appropiete. Which parameters would you use for called initialize? And how do you access those parameters? I prefer the approach of properly setting things up wiuth initialize() and end the execution otherwise. 
4. True. Added.